ibis/backends/base/sql/alchemy/__init__.py

```diff
@@ -749,10 +728,28 @@ def _compile_pyarrow_udf(self, udf_node: ops.ScalarUDF) -> str:
                 f"The {self.name} backend does not support PyArrow-based vectorized scalar UDFs"
             )
 
+    def _define_udf_translation_rules(self, expr):
```

Comment:
I find this implementation fairly complicated, we should refactor it soon enough.